atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/config/OracleConnectionConfig.java

```diff
@@ -250,4 +255,30 @@ public interface ServiceNameConfiguration {
 
         class Builder extends ImmutableServiceNameConfiguration.Builder {}
     }
+
+    @Override
+    public String getSqlExceptionOverrideClassname() {
+        return OracleSqlExceptionOverride.class.getName();
+    }
+
+    public static class OracleSqlExceptionOverride implements SQLExceptionOverride {
+
+        // Must be public as Hikari instantiates via reflection
+        public OracleSqlExceptionOverride() {}
+
+        /*
+         * See https://github.com/brettwooldridge/HikariCP/pull/1838
+         * HikariCP introduced a "bug" in https://github.com/brettwooldridge/HikariCP/issues/1308
+         * which treats QueryTimeout as a connection close + evict event.
+         * This is not correct - SQLTimeoutException inherits from SQLTransientException
+         * and therefore Hikari should not close the connection. We have existing Gotham tests that assert this
+         * behaviour.
+         */
+        @java.lang.Override
+        public SQLExceptionOverride.Override adjudicate(SQLException sqlException) {
+            return sqlException instanceof SQLTimeoutException
+                    ? SQLExceptionOverride.Override.DO_NOT_EVICT
+                    : SQLExceptionOverride.Override.CONTINUE_EVICT;
```

Comment:
Do we have test coverage showing this does in fact get hit? Judging from comment https://github.com/brettwooldridge/HikariCP/issues/1388#issuecomment-1047874286 this callback may not work with hikari 5.0.1+

Seems like https://github.com/brettwooldridge/HikariCP/blob/dev/src/main/java/com/zaxxer/hikari/pool/ProxyConnection.java#L154-L163 is the code intended to handle these cases, should ensure this callback is actually invoked. Can we simulate via unit test?

Reply:
Not sure which comment you are referring to. Building Gotham locally with a local Atlas build including this change (not evicting on SQLTimeoutException), Gothams SQLTest passes again. Without it, and with Gotham pointing to a newer Hikari it fails.

Reply:
Oh I see "@idkw Is your workaround still valid for version 5.0.1? I set it using HikariConfig#setExceptionOverrideClassName, but it doesn't seem to get triggered." Not sure - it definitely is triggered.